pipeline-model-definition/pom.xml

```diff
@@ -120,7 +120,7 @@
     <dependency>
       <groupId>org.jenkins-ci.plugins</groupId>
       <artifactId>git</artifactId>
-      <scope>test</scope>
+      <optional>true</optional>
```

Comment:
Ok, now my head is exploding. `target/test-classes/test-dependencies` is identical whether I'm running `mvn clean test -Dtest="BasicModelDefTest#folderLibraryParsing"` with `<optional>true</optional>` or `<scope>test</scope>`, but the test is consistently failing with `<optional>true</optional>` (with the fix in `Converter.groovy` commented out) and passing with `<scope>test</scope>`.

I am very, very mystified.